docs: add information on how to type custom hooks (#22312)
nuxt/nuxt: docs/2.guide/3.going-further/2.hooks.md

```js [plugins/test.ts]
export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.hook('page:start', () => {

>       /* your code goes here */
>     })

Reviewer: ```suggestion
    /* your code goes here */
  })
```

::alert{icon=👉}
Learn more about available [Nitro lifecycle hooks](/docs/api/advanced/hooks#nitro-app-hooks-runtime-server-side).
::

## Add additional hooks

> You can add additional hooks by overloading types provided by Nuxt. This can be useful for modules.

Reviewer: ```suggestion
You can add additional hooks by augmenting the types provided by Nuxt. This can be useful for modules.
```

```ts
import { HookResult } from "@nuxt/schema";

declare module '#app' {

>     interface RuntimeNuxtHooks {
>         "your-nuxt-runtime-hook": () => HookResult
>     }
>     interface NuxtHooks {
>         "your-nuxt-hook": () => HookResult
>     }
> }
> 
> declare module 'nitropack' {
>     interface NitroRuntimeHooks {
>         'your-nitro-hook': () => void;
>     }
> }

Reviewer: ```suggestion
  interface RuntimeNuxtHooks {
    'your-nuxt-runtime-hook': () => HookResult
  }
  interface NuxtHooks {
    'your-nuxt-hook': () => HookResult
  }
}

declare module 'nitropack' {
  interface NitroRuntimeHooks {
    'your-nitro-hook': () => void;
  }
}
```
